src/clib/pio_darray.c

```diff
@@ -91,16 +95,19 @@ int PIOc_write_darray_multi(const int ncid, const int *vid, const int ioid,
     if (!(iodesc = pio_get_iodesc_from_id(ioid)))
         return PIO_EBADID;
 
+    /* ??? */
     vdesc0 = file->varlist + vid[0];
 
-    //   rlen = iodesc->llen*nvars;
+    /* ??? */
+    /*   rlen = iodesc->llen*nvars; */
     rlen=0;
     if (iodesc->llen > 0)
         rlen = iodesc->maxiobuflen*nvars;
 
     if (vdesc0->iobuf)
         piodie("Attempt to overwrite existing io buffer",__FILE__,__LINE__);
 
+    /* ??? */
     if (iodesc->rearranger > 0)
```

Comment:
I originally conceived of an option in which the data could be provided to pio already in an arrangement suitable for passing on to the underlying io layer, but I've yet to find a case where that is true and so rearranger is always > 0.   Currently there are two rearrangers box=1 and subset=2.   There is never a case where rearranger==0